R/utils.R

```diff
@@ -58,8 +58,13 @@ include_parent_datanames <- function(dataname, join_keys) {
 #' @keywords internal
 teal_data_to_filtered_data <- function(x, datanames = teal.data::datanames(x)) {
   checkmate::assert_class(x, "teal_data")
-  checkmate::assert_character(datanames, min.len = 1L, any.missing = FALSE)
-  checkmate::assert_subset(datanames, teal.data::datanames(x))
+  checkmate::assert_character(datanames, any.missing = FALSE)
+
+  if (length(datanames) == 0) {
```

Comment:
@kartikeyakirar I think you need a new internal teal function to extract datanames from `teal_data` object

```r
#' Wrapper on `teal.data::datanames`
#'
#' Special function used in internals of `teal` to return names of datasets even if `datanames` 
#' has not been set. 
#' @param data (`teal_data`)
#' @return `character`
# @keywords internal
teal_data_datanames <- function(data) {
  checkmate::assert_class(data, "teal_data")
  if (length(teal.data::datanames(data))) {
    teal.data::datanames(data)
  } else {
    warning("`data` object has no datanames. Default datanames are set using `teal_data`'s environment.")
    ls(data@env)
  }
}
```

This function should probably replace all `teal.data::datanames()` calls in `teal`. At this moment solution is insufficient as most of the asserts will fail (please check teal.gallery/exploratory app for example).

Reply:
replacing` teal.data::datanames` call  for` teal`  this will probably gives many warnings from different places.

Reply:
> replacing teal.data::datanames call for teal this will probably gives many warnings from different places.

Let's remove warning from the function and warn once in `srv_teal_with_splash` just before `is_modules_ok <- ...`:

```r
if (!length(teal.data::datanames(data))) {
  warning("`data` object has no datanames. Default datanames are set using `teal_data`'s environment.")
}

```

Reply:
added suggested changes here [3504669](https://github.com/insightsengineering/teal/pull/1004/commits/3504669b005a2ebdd2687259072d7e9aae587b59)